modules/packages/MPI.chpl

```diff
@@ -0,0 +1,370 @@
+  could be more specific.
+  */
+  extern const MPI_BOTTOM : opaque;
+  extern const MPI_PROC_NULL : opaque;
+  extern const MPI_ANY_SOURCE : opaque;
+  extern const MPI_ANY_TAG : opaque;
+  extern const MPI_UNDEFINED : opaque;
+  extern const MPI_BSEND_OVERHEAD : opaque;
+  extern const MPI_KEYVAL_INVALID : opaque;
+
+  /* Error handling specifiers */
+  extern const MPI_ERRORS_ARE_FATAL : MPI_Errhandler;
+  extern const MPI_ERRORS_RETURN : MPI_Errhandler;
+
+  /* Maximum sizes for strings. 
+  TODO: Are these correctly defined?
+  */
+  extern const MPI_MAX_PROCESSOR_NAME : c_int;
+  extern const MPI_MAX_ERROR_STRING : c_int;
+
+  /* Elementary datatypes */
+  extern const MPI_CHAR : MPI_Datatype;
+  extern const MPI_SHORT : MPI_Datatype;
+  extern const MPI_INT   : MPI_Datatype;
+  extern const MPI_LONG  : MPI_Datatype;
+  extern const MPI_UNSIGNED_CHAR : MPI_Datatype;
+  extern const MPI_UNSIGNED_SHORT : MPI_Datatype;
+  extern const MPI_UNSIGNED : MPI_Datatype;
+  extern const MPI_UNSIGNED_LONG : MPI_Datatype;
+  extern const MPI_FLOAT  : MPI_Datatype;
+  extern const MPI_DOUBLE  : MPI_Datatype;
+  extern const MPI_LONG_DOUBLE : MPI_Datatype;
+  extern const MPI_BYTE  : MPI_Datatype;
+  extern const MPI_PACKED : MPI_Datatype;
+
+  /* Datatypes for reductions */
+  extern const MPI_FLOAT_INT : MPI_Datatype;
+  extern const MPI_DOUBLE_INT : MPI_Datatype;
+  extern const MPI_LONG_INT : MPI_Datatype;
+  extern const MPI_2INT : MPI_Datatype;
+  extern const MPI_SHORT_INT : MPI_Datatype;
+  extern const MPI_LONG_DOUBLE_INT : MPI_Datatype;
+
+  /* Optional datatypes */
+  extern const MPI_LONG_LONG_INT : MPI_Datatype;
+
+  /* Special datatypes for constructing derived datatypes */
+  extern const MPI_UB : MPI_Datatype;
+  extern const MPI_LB : MPI_Datatype;
+
+  /* Reserved communicators */
+  extern const MPI_COMM_WORLD : MPI_Comm;
+  extern const MPI_COMM_SELF : MPI_Comm;
+
+  /* Communicator/ Group comparisons */
+  // These all appear to be C integers in MPI code
+  extern const MPI_IDENT : c_int;
+  extern const MPI_CONGRUENT : c_int;
+  extern const MPI_SIMILAR : c_int;
+  extern const MPI_UNEQUAL : c_int;
+
+  /* Environmental inquiry keys */ 
+  // All of the above appear to be C enums or defines....
+  extern const MPI_TAG_UB : c_int; 
+  extern const MPI_IO : c_int; 
+  extern const MPI_HOST : c_int; 
+  extern const MPI_WTIME_IS_GLOBAL : c_int; 
+
+  /* Collective operations */
+  extern const MPI_MAX : MPI_Op;
+  extern const MPI_MIN : MPI_Op;
+  extern const MPI_SUM : MPI_Op;
+  extern const MPI_PROD : MPI_Op;
+  extern const MPI_MAXLOC : MPI_Op;
+  extern const MPI_MINLOC : MPI_Op;
+  extern const MPI_BAND : MPI_Op;
+  extern const MPI_BOR : MPI_Op;
+  extern const MPI_BXOR : MPI_Op;
+  extern const MPI_LAND : MPI_Op;
+  extern const MPI_LOR : MPI_Op;
+  extern const MPI_LXOR : MPI_Op;
+
+  /*
+   C-API 
+
+   We wrap all of these into a ``C_MPI`` submodule,
+   since we likely will add in some helper routines
```

Comment:
I'm assuming you havn't added in the helper routines yet, right? I think it would be nice to note that this is future work in the PR message.

It sure would be nice to have Chapel versions of some of the functions that used the compile-time type information to get the datatypes like MPI_INT...